### PR TITLE
Upmerge + Remove audio board files from sdk_nrf

### PR DESCRIPTION
**Pull request by koffes**
This change is dependent on upstream: https://github.com/zephyrproject-rtos/zephyr/pull/55639

### CODE REVIEW

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 1]
```diff
@@ -160,7 +160,7 @@ static void stream_stopped_cb(struct bt_audio_stream *stream)
 	LOG_INF("Stream index %d stopped", active_stream_index);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:160]:
```suggestion
	LOG_INF("Stream index %d stopped, reason: %d", active_stream_index, reason);
```

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c [hunk 2]
```diff
@@ -347,17 +350,17 @@ static void stream_started_cb(struct bt_audio_stream *stream)
...
 }
 
-static void stream_disabled_cb(struct bt_audio_stream *stream)
+static void stream_disabled_cb(struct bt_bap_stream *stream)
 {
 	LOG_DBG("Audio Stream %p disabled", (void *)stream);
 }
 
-static void stream_stopped_cb(struct bt_audio_stream *stream)
+static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:363]: Add reason to LOG_INF

File: applications/nrf5340_audio/src/modules/hw_codec.c [hunk 3]
```diff
@@ -58,27 +55,6 @@ static int cs47l63_comm_reg_conf_write(const uint32_t config[][2], uint32_t num_
...
-		return -ENXIO;
-	}
-
-	ret = gpio_pin_configure_dt(&hw_codec_sel, GPIO_OUTPUT_LOW);
-	if (ret) {
-		return ret;
-	}
-
-	/* Allow for switches to flip when selecting on board hw_codec */
-	k_msleep(HW_CODEC_SELECT_DELAY_MS);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/modules/hw_codec.c:77]: This is no longer needed?

**Review by alexsven — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/audio/Kconfig [hunk 4]
```diff
@@ -32,8 +32,8 @@ config AUDIO_FRAME_DURATION_US
 
 choice AUDIO_SAMPLE_RATE
 	prompt "Audio sample rate"
-	default AUDIO_SAMPLE_RATE_16000_HZ if BT_AUDIO_BROADCAST_16_2_1 || BT_AUDIO_BROADCAST_16_2_2 || BT_AUDIO_UNICAST_16_2_1
-	default AUDIO_SAMPLE_RATE_24000_HZ if BT_AUDIO_BROADCAST_24_2_1 || BT_AUDIO_BROADCAST_24_2_2 || BT_AUDIO_UNICAST_24_2_1
+	default AUDIO_SAMPLE_RATE_16000_HZ if BT_AUDIO_BROADCAST_16_2_1 || BT_AUDIO_BROADCAST_16_2_2 || BT_BAP_UNICAST_16_2_1
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/audio/Kconfig:35]: Like @alexsven mentioned, here only unicast is changed

Comment (erikrobstad) [reply applications/nrf5340_audio/src/audio/Kconfig:35]: Looks like broadcast is not changed several places

File: applications/nrf5340_audio/src/bluetooth/Kconfig [hunk 5]
```diff
@@ -230,29 +230,29 @@ config BLE_ISO_RX_STATS_S
 	int "Interval in seconds to print Bluetooth LE ISO RX stats. 0 to deactivate"
 	default 0
 
-choice BT_AUDIO_UNICAST_BAP_CONFIGURATION
+choice BT_BAP_UNICAST_BAP_CONFIGURATION
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/Kconfig:233]: Hmm, I think maybe the best naming scheme could be:
```suggestion
choice LE_AUDIO_UNICAST_BAP_CONFIGURATION
```
Our .c/.h files are called le_audio_(...)

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults [hunk 6]
```diff
@@ -15,6 +15,12 @@ config BT_ATT_ENFORCE_FLOW
 config BT_HCI_VS_EXT
 	default n
 
+config BT_HCI_VS
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/Kconfig.defaults:18]: These are just some new configs that are needed for the latest Zephyr version?

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 5 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/Kconfig:233]: In other words: BT_AUDIO -> LE_AUDIO

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults [hunk 7]
```diff
@@ -246,13 +253,14 @@ config BT_DEVICE_APPEARANCE
...
 
-config BT_AUDIO_BROADCAST_SOURCE
+config BT_BAP_BROADCAST_SOURCE
 	default y
 
 config BT_ISO_TX_BUF_COUNT
 	default 2
 
 config BT_AUDIO_BROADCAST_SRC_STREAM_COUNT
+	int "Number of broadcast source streams"
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/Kconfig.defaults:263]: Is there a specific reason for adding the text on this one (and for BT_AUDIO_BROADCAST_NAME)?
There are several other configs without text.
This is a .defaults file, meaning that the configs should be explained in the Kconfig file where they are defined

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c [hunk 8]
```diff
@@ -8,9 +8,11 @@
 
 #include <zephyr/bluetooth/bluetooth.h>
 #include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/bap_lc3_preset.h>
 /* TODO: Remove when a get_info function is implemented in host */
-#include <../subsys/bluetooth/audio/endpoint.h>
-#include <../subsys/bluetooth/audio/audio_iso.h>
+#include <../subsys/bluetooth/audio/bap_endpoint.h>
+#include <../subsys/bluetooth/audio/bap_iso.h>
+
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c:15]: Remove newline

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c [hunk 9]
```diff
@@ -9,8 +9,11 @@
 #include <zephyr/bluetooth/bluetooth.h>
 #include <zephyr/bluetooth/audio/audio.h>
 #include <zephyr/bluetooth/hci.h>
-#include <../subsys/bluetooth/audio/endpoint.h>
-#include <../subsys/bluetooth/audio/audio_iso.h>
+#include <zephyr/bluetooth/audio/bap.h>
+
+#include <zephyr/bluetooth/audio/bap_lc3_preset.h>
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:14]: The newline should be after
#include <zephyr/bluetooth/audio/bap_lc3_preset.h>
and not before?

File: applications/nrf5340_audio/src/drivers/cs47l63_comm.c [hunk 10]
```diff
@@ -27,15 +29,16 @@ LOG_MODULE_REGISTER(CS47L63, CONFIG_CS47L63_LOG_LEVEL);
 #define CS47L63_PROCESS_THREAD_DELAY_MS 10
 
 static const struct gpio_dt_spec hw_codec_gpio =
-	GPIO_DT_SPEC_GET(DT_NODELABEL(hw_codec_gpio_in), gpios);
+	GPIO_DT_SPEC_INST_GET(0, gpio9_gpios);
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/drivers/cs47l63_comm.c:32]: So now the HW codec pins are not defined in the board files?

File: applications/nrf5340_audio/src/utils/board_version.c [hunk 11]
```diff
@@ -18,56 +18,41 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(board_version, CONFIG_BOARD_VERSION_LOG_LEVEL);
 
-#define ADC_1ST_CHANNEL_ID 0
-#define ADC_RESOLUTION_BITS 12
+#define BOARD_ID DT_NODELABEL(board_id)
+
+static const struct adc_dt_spec adc = ADC_DT_SPEC_GET(BOARD_ID);
+static const struct gpio_dt_spec power_gpios = GPIO_DT_SPEC_GET(BOARD_ID, power_gpios);
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/utils/board_version.c:24]: What does "power_gpios" mean in this context?

**Review by erikrobstad — CHANGES_REQUESTED** [review]
I have not enough background knowledge to review the HW codec changes thoroughly

File: applications/nrf5340_audio/src/utils/board_version.c (hunk 11 above)
Comment (jfischer-no) [reply applications/nrf5340_audio/src/utils/board_version.c:24]: https://docs.zephyrproject.org/latest/build/dts/api/bindings/adc/voltage-divider.html

https://github.com/zephyrproject-rtos/zephyr/pull/55675/commits/f9f135531b4e124c8567181f7ea4c0320d1f6f5b

**Comment by gmarull** [comment]
This PR should be squashed and integrated in the upmerge PR, right?

File: applications/nrf5340_audio/src/drivers/cs47l63_comm.c (hunk 10 above)
Comment (koffes) [reply applications/nrf5340_audio/src/drivers/cs47l63_comm.c:32]: They are, just in a different location: https://github.com/zephyrproject-rtos/zephyr/pull/55675

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c (hunk 1 above)
Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:160]: Done

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c (hunk 2 above)
Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:363]: Done

File: applications/nrf5340_audio/src/modules/hw_codec.c (hunk 3 above)
Comment (koffes) [reply applications/nrf5340_audio/src/modules/hw_codec.c:77]: Yes, this is now done much earlier by Zephyr. Check out `gpio-hogs`

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c (hunk 8 above)
Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c:15]: Done

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c (hunk 9 above)
Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:14]: Correct. Done

**Comment by koffes** [comment]
Depending on https://github.com/zephyrproject-rtos/zephyr/pull/56121 and https://github.com/zephyrproject-rtos/zephyr/pull/54111

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 5 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/Kconfig:239]: What is the reasoning for keeping the BAP name scheme?

File: applications/nrf5340_audio/src/audio/Kconfig (hunk 4 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/audio/Kconfig]: Did it compile before changing the broadcast configs to BAP?

File: applications/nrf5340_audio/src/audio/streamctrl.c [hunk 12]
```diff
@@ -124,12 +124,12 @@ static void le_audio_rx_data_handler(uint8_t const *const p_data, size_t data_si
 	struct ble_iso_data *iso_received = NULL;
 
 #if (CONFIG_AUDIO_DEV == GATEWAY)
+	static uint32_t packet_count_r;
 	switch (channel_index) {
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/audio/streamctrl.c:128]: Is it necessary to move this?
If so, it should be a newline before the switch

File: applications/nrf5340_audio/src/bluetooth/Kconfig [hunk 13]
```diff
@@ -230,29 +230,35 @@ config BLE_ISO_RX_STATS_S
 	int "Interval in seconds to print Bluetooth LE ISO RX stats. 0 to deactivate"
 	default 0
 
-choice BT_AUDIO_UNICAST_BAP_CONFIGURATION
+config CODEC_CAPAB_COUNT_MAX
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/Kconfig:233]:
```suggestion
config CODEC_CAP_COUNT_MAX
```

I think CODEC_CAP is used most places

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c [hunk 14]
```diff
@@ -739,8 +751,10 @@ static void discover_sink_cb(struct bt_conn *conn, struct bt_codec *codec, struc
 	}
 
 	/* At this point the location/channel index of the headset is always known */
-	memcpy(headsets[channel_index].sink_codec_cap, temp_codec_cap[temp_cap_index].cap,
-	       sizeof(temp_codec_cap[temp_cap_index].cap));
+	for (int i = 0; i < CONFIG_CODEC_CAPAB_COUNT_MAX; i++) {
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:754]: Hmm, why are these changes here in this PR?
This is changes from another sdk-nrf PR/commit

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c [hunk 15]
```diff
@@ -14,7 +14,9 @@
 #include <zephyr/bluetooth/audio/pacs.h>
 #include <zephyr/bluetooth/audio/csip.h>
 #include <zephyr/bluetooth/audio/cap.h>
-#include <../subsys/bluetooth/audio/endpoint.h>
+
+/* TODO: Remove when a get_info function is implemented in host */
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:18]: Should have the same comment all places where this file is included

File: applications/nrf5340_audio/src/drivers/cs47l63_comm.c (hunk 10 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/drivers/cs47l63_comm.c:31]: Aha, what does the name "gpio9_gpios" mean? To me it sounded like a generic GPIO define

**Review by erikrobstad — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/drivers/cs47l63_comm.c (hunk 10 above)
Comment (jfischer-no) [reply applications/nrf5340_audio/src/drivers/cs47l63_comm.c:31]: https://github.com/zephyrproject-rtos/zephyr/blob/4d9e13115140abb4a86dda3b4a5c7145a20949cd/dts/bindings/sound/cirrus%2Ccs47l63.yaml#L64-L67

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c (hunk 14 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:754]: Nevermind, this should be correct :)

File: applications/nrf5340_audio/src/audio/streamctrl.c (hunk 12 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/audio/streamctrl.c:128]: Certain versions of the compiler complains, and I believe rightly so, if a variable is declared within a switch case. We either need to move the declaration or add { in the case

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 13 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/Kconfig:233]: I think it was changed because it can be confused with Common Audio Profile. But since it doesn't have the prefix BT_AUDIO it should be fine to change to CAP in this case.

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 5 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/Kconfig:239]: I believe that is to match the renaming in host. We should have a separate task for changing all our configs to LE_AUDIO_... perhaps?

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c [hunk 16]
```diff
@@ -664,16 +710,28 @@ static int initialize(le_audio_receive_cb recv_cb, le_audio_timestamp_cb timestm
...
 	if (ret) {
-		LOG_ERR("Available context set failed");
+		LOG_ERR("Available context set failed. Err: %d", ret);
 		return ret;
 	}
 #endif /* CONFIG_STREAM_BIDIRECTIONAL */
 
 	for (int i = 0; i < ARRAY_SIZE(audio_streams); i++) {
-		bt_audio_stream_cb_register(&audio_streams[i], &stream_ops);
+			bt_bap_stream_cb_register(&audio_streams[i], &stream_ops);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:734]: Seems to be an extra tab here? Might just be github

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c [hunk 17]
```diff
@@ -455,16 +454,41 @@ static void stream_recv_cb(struct bt_audio_stream *stream, const struct bt_iso_r
...
+	}
 }
 
-static void stream_stop_cb(struct bt_audio_stream *stream)
+static void stream_stop_cb(struct bt_bap_stream *stream, uint8_t reason)
 {
 	int ret;
 
-	LOG_DBG("Stream %p stopping", stream);
+	LOG_INF("Stream stopped. Reason: %d", reason);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:491]: I think this should be DBG. The INF should be in streamctrl, when LE_AUDIO_EVT_NOT_STREAMING is received.

Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:491]: Alternatively, the reason could be DBG

Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:491]: I also see that we have callbacks for changes to the stream, but also callbacks for changes to the unicast_server. I think we need to go through and decide which one of them to use, as I think they are both called most of the times. Maybe @Thalley can provide more information.

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c (hunk 15 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:18]: Probably yes

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 18]
```diff
@@ -151,17 +151,17 @@ static void stream_started_cb(struct bt_audio_stream *stream)
...
-static void stream_stopped_cb(struct bt_audio_stream *stream)
+static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
 {
 	int ret;
 
 	ret = ctrl_events_le_audio_event_send(LE_AUDIO_EVT_NOT_STREAMING);
 	ERR_CHK(ret);
 
-	LOG_INF("Stream index %d stopped", active_stream_index);
+	LOG_INF("Stream index %d stopped. Reason: %d", active_stream_index, reason);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:161]: I think the reason should be DBG

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c [hunk 19]
```diff
@@ -118,17 +119,17 @@ static void stream_started_cb(struct bt_audio_stream *stream)
...
-static void stream_stopped_cb(struct bt_audio_stream *stream)
+static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
 {
 	int ret;
 
 	ret = ctrl_events_le_audio_event_send(LE_AUDIO_EVT_NOT_STREAMING);
 	ERR_CHK(ret);
 
-	LOG_INF("Broadcast source %p stopped", (void *)stream);
+	LOG_INF("Broadcast source %p stopped. Reason: %d", (void *)stream, reason);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c:129]: The reason should be DBG

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c [hunk 20]
```diff
@@ -424,8 +425,8 @@ static void stream_qos_set_cb(struct bt_audio_stream *stream)
 		LOG_DBG("Set %s to PD: %d", headsets[channel_index].ch_name, stream->qos->pd);
 
 		if (playing_state) {
-			ret = bt_audio_stream_enable(stream, lc3_preset_sink.codec.meta,
-						     lc3_preset_sink.codec.meta_count);
+		ret = bt_bap_stream_enable(stream, lc3_preset_sink.codec.meta,
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:428]: Is this no longer indented correctly, or is this GitHub -trickery again?

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c [hunk 21]
```diff
@@ -506,22 +507,22 @@ static void stream_started_cb(struct bt_audio_stream *stream)
...
 }
 
-static void stream_stopped_cb(struct bt_audio_stream *stream)
+static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
 {
 	int ret;
 	uint8_t channel_index;
 
-	LOG_INF("Stream %p stopped", (void *)stream);
+	LOG_INF("Stream %p stopped. Reason %d", (void *)stream, reason);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:525]: Reason should be DBG, and perhaps the pointer to the stream. It could be better to print the channel_index actually, but willing to discuss

**Review by alexsven — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 5 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/Kconfig:239]: Yes, we can discuss this in the architecture discussions

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 13 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/Kconfig:233]: Since it says CODEC before, it should be fine to call it CAP

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c (hunk 16 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:734]: Hmmm, several of the files are not clang-formatted

Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:734]: I have gone through all files and clang-formatted now

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c (hunk 20 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:428]: I have gone through all files and clang-formatted now

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c (hunk 17 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c:491]: As discussed on Teams, this will be changed when we do the architecture rewrite

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c (hunk 18 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:161]: As discussed on Teams, this will be addressed when we do the architecture rewrite

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c (hunk 19 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c:129]: As discussed on Teams, this will be addressed when we do the architecture rewrite

File: applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c (hunk 21 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c:525]: As discussed on Teams, this will be addressed when we do the architecture rewrite

**Review by alexsven — APPROVED** [review]

**Comment by erikrobstad** [comment]
Rebased, since we needed the controller hex move commit

**Comment by jfischer-no** [comment]
cherry-picked in https://github.com/nrfconnect/sdk-nrf/pull/10535
